mlflow/pyfunc/model.py

```diff
@@ -327,12 +333,40 @@ def _load_pyfunc(model_path):
         python_model = cloudpickle.load(f)
 
     artifacts = {}
+    hf_prefix = "hf:/"
     for saved_artifact_name, saved_artifact_info in pyfunc_config.get(
         CONFIG_KEY_ARTIFACTS, {}
     ).items():
-        artifacts[saved_artifact_name] = os.path.join(
-            model_path, saved_artifact_info[CONFIG_KEY_ARTIFACT_RELATIVE_PATH]
-        )
+        if CONFIG_KEY_ARTIFACT_RELATIVE_PATH in saved_artifact_info:
+            artifacts[saved_artifact_name] = os.path.join(
+                model_path, saved_artifact_info[CONFIG_KEY_ARTIFACT_RELATIVE_PATH]
+            )
+        else:
+            artifact_uri = saved_artifact_info[CONFIG_KEY_ARTIFACT_URI]
+            if artifact_uri.startswith(hf_prefix):
+                try:
+                    from huggingface_hub import snapshot_download
+                except ImportError as e:
+                    raise MlflowException(
+                        "Failed to import huggingface_hub. Please install huggingface_hub to "
+                        f"download snapshot with artifact_uri: {artifact_uri}. Error: {e}"
+                    )
+                try:
+                    hf_uri = artifact_uri[len(hf_prefix) :]
+                    splits = hf_uri.split("?")
+                    repo_id = splits[0]
+                    kwargs = (
+                        dict([param.split("=") for param in splits[1].split("&")])
+                        if len(splits) > 1
+                        else {}
+                    )
+                    snapshot_location = snapshot_download(repo_id=repo_id, **kwargs)
+                    artifacts[saved_artifact_name] = snapshot_location
+                except Exception as e:
+                    raise MlflowException.invalid_parameter_value(
+                        "Failed to download snapshot from Hugging Face Hub with "
+                        f"artifact_uri: {artifact_uri}. Error: {e}"
+                    )
```

Comment:
```suggestion
                    snapshot_location = snapshot_download(repo_id=repo_id, **kwargs)
                except Exception as e:
                    raise MlflowException.invalid_parameter_value(
                        "Failed to download snapshot from Hugging Face Hub with "
                        f"artifact_uri: {artifact_uri}. Error: {e}"
                    )
                artifacts[saved_artifact_name] = snapshot_location
```

We can pull out the lines that are unlikely to fail from try-catch.

Reply:
Example:

```
try:
    a()
    b()
    c()
except Exception:
    ...
```

What's the expectation here? So `a`, `b`, and `c` can fail?

```
try:
    a()
except Exception:
    ...

b()
c()
```

`a()` can fail.

Reply:
Make sense, I'll update after we decide on which approach to take :)

Reply:
Sounds good!